x-pack/legacy/plugins/maps/public/layers/util/pushed_data.js

```diff
@@ -0,0 +1,21 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+export class PushedData {
```

Comment:
Not sure we need a class for `PushedData`. Maybe just the methods `hasPushedData` and `getPushedData` methods to `AbstractLayer`?

Reply:
We definitely could, but I think it's easier to have an analog for DataRequests. So there's a class for data that's dynamically requested, and a class for data that's pushed to our application.

> How about renaming pushedData? Maybe cannedData or staticData or injectedData or inlineData? The work "pushed" is a little vague.

Agreed. I like `InjectedData`, I'll update it!